### PR TITLE
JSON reloaded

### PR DESCRIPTION
**Pull request by rturrado**
C++ code strings can now be written:
- in a one-string block, without so many quotes and new line characters, and without indentations,
- thanks to the combined use of `fmt::print` and string literal suffix operators.

I also took the opportunity to add `gtest` to the project, and combine all the tests into a single one.

Due to some new code making use of the `range-v3` library:
- the `range-v3` dependency has been added, and
- the project's C++ standard updated to C++17.

`tree-gen-cpp`:
- Changed to use the string literal suffix operators.
- Used nested namespaces.

`format_utils`:
- Added `generator/format_utils.{hpp,cpp}` to contain the string literal suffix operators implementations.
- Added `test/test_format_utils.cpp`.

Added `gtest` to the project:
- Added `test/main.cpp`.
- Changed `test_annotatable.cpp` and `test_cbor.cpp` to be `gtest`s.
- Removed `test/assert.hpp`.
- Renamed `tests` folder to `test`.

Added JSON checks in the 'directory' and 'interpreter' examples.

### CODE REVIEW

File: test/test_annotatable.cpp [hunk 1]
```diff
@@ -1,7 +1,8 @@
-#include <sstream>
-#include <cstdio>
 #include "tree-annotatable.hpp"
-#include "assert.hpp"
+
+#include <gmock/gmock.h>
```
Comment (pablolh) [inline test/test_annotatable.cpp]: Is there not just a `gtest/gtest.h` header?

File: generator/format_utils.cpp [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+namespace tree_gen::cpp {
+
+/**
+ * Formats string literal containing C++ code.
+ *
+ * string_literal is the input string.
+ * indent is the indentation, in number of spaces, of the output string.
+ * remove_first_new_line will be true if we want the first new line of the input string to be removed, false otherwise.
+ */
+std::string format_string_literal(std::string_view string_literal, size_t indent, bool remove_first_line) {
```
Comment (pablolh) [inline generator/format_utils.cpp]: As dicussed would be nice to do some preprocessing here to avoid having to escpae { as {{ all the time :)

Comment (rturrado) [reply generator/format_utils.cpp]: Done. I can't believe this took me all day haha.

File: test/test_annotatable.cpp (hunk 1 above)
Comment (rturrado) [reply test/test_annotatable.cpp]: Thanks for the comment, Pablo! I've just removed the `gmock` dependency. This is one of the dangers of the copy-and-paste programming! hehe.

**Review by pablolh — APPROVED** [review]

**Pull request merged**